src/v1/sources/shopify/pixelTransform.js

```diff
@@ -21,6 +30,60 @@ const NO_OPERATION_SUCCESS = {
   statusCode: 200,
 };
 
+/**
+ * Parses and extracts cart token value from the input event
+ * @param {Object} inputEvent
+ * @returns {String} cartToken
+ */
+function extractCartToken(inputEvent) {
+  const cartTokenLocation = pixelEventToCartTokenLocationMapping[inputEvent.name];
+  if (!cartTokenLocation) {
+    stats.increment('shopify_pixel_cart_token_not_found', {
+      event: inputEvent.name,
+      writeKey: inputEvent.query_parameters.writeKey,
+    });
+    return undefined;
+  }
+  // the unparsedCartToken is a string like '/checkout/cn/1234'
+  const unparsedCartToken = _.get(inputEvent, cartTokenLocation);
+  if (typeof unparsedCartToken !== 'string') {
+    logger.error(`Cart token is not a string`);
+    stats.increment('shopify_pixel_cart_token_not_found', {
+      event: inputEvent.name,
+      writeKey: inputEvent.query_parameters.writeKey,
+    });
+    return undefined;
+  }
+  const cartTokenParts = unparsedCartToken.split('/');
+  const cartToken = cartTokenParts[3];
+  return cartToken;
+}
+
+/**
+ * Handles storing cart token and anonymousId (clientId) in Redis, and updating the event with the anonymousId
+ * @param {Object} inputEvent
+ * @param {String} clientId
+ */
+const handleCartTokenRedisOperations = async (inputEvent, clientId) => {
+  const cartToken = extractCartToken(inputEvent);
+  try {
+    if (isDefinedNotNullNotEmpty(clientId) && isDefinedNotNullNotEmpty(cartToken)) {
+      await RedisDB.setVal(cartToken, ['anonymousId', clientId]);
+      stats.increment('shopify_pixel_cart_token_set', {
+        event: inputEvent.name,
+        writeKey: inputEvent.query_parameters.writeKey,
+      });
+    }
+    inputEvent.anonymousId = clientId;
```

Comment:
for events where we are not doing the redis operations (i.e where there is no cart token), we are adding the clientId as anonymousId there, otherwise they are added inside the redis operation functions.

Reply:
I don't think so, it's outside the switch case - getting applied to all events

Reply:
Yes yes, it is redundant to add inside